AirCastingTests/Logic/SDSync/SdSyncAveragingServiceTests.swift

```diff
@@ -0,0 +1,135 @@
+// Created by Lunar on 13/10/2022.
+//
+
+import XCTest
+@testable import AirCasting
+
+extension SDSyncMeasurement: Equatable {
+    public static func == (lhs: AirCasting.SDSyncMeasurement, rhs: AirCasting.SDSyncMeasurement) -> Bool {
+        lhs.measuredAt == rhs.measuredAt && lhs.value == rhs.value && lhs.location?.latitude == rhs.location?.latitude && lhs.location?.longitude == rhs.location?.longitude
+    }
+}
+
+final class SdSyncAveragingServiceTests: XCTestCase {
+    let service = DefaultSDSyncAveragingService()
+
+    func test_averageMeasurementsWithReminderCalled_returnsUnaveragedMeasurementsAsReminder() throws {
+        let date = Date()
+        
+        let measurements = [SDSyncMeasurement(measuredAt: date, value: 3)]
+        
+        let reminder = service.averageMeasurementsWithReminder(measurements: measurements, startTime: date, averagingWindow: .firstThresholdWindow, action: { _, _ in })
+        XCTAssertEqual(reminder, measurements)
+    }
+    
+    func test_averageMeasurementsWithReminderCalled_callsCompletionWithAnAveragedMeasurement() throws {
```

Comment:
maybe `finished` instead of `callsCompletion` as it sounds let's say low level?

Reply:
But we want to check if the completion gets called. I think finished doesn't suggest that.